### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by lgiussan**

### CODE REVIEW

**Review by 95DDB — APPROVED** [review]

File: app/src/app.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    // Listen for the 'finalized' event
+    events.on(ZkVerifyEvents.Finalized, ({ blockHash }) => {
+        console.log(`Transaction finalized in zkVerify, block-hash: ${blockHash}`);
+    });
+
+    // Handle errors during the transaction process
+    events.on('error', (error) => {
+        console.error('An error occurred during the transaction:', error);
+    });
```
Comment (95DDB) [inline app/src/app.js:65]: Perhaps we can be explicit here about the type of events/errors and events associated to success/errors ? Just for "documentation" purposes, no need to handle them explicitly here

File: app/src/app.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    let merkleProof, numberOfLeaves, leafIndex;
+    try {
+        const proofDetails = await session.poe(attestationId, leafDigest);
+        ({ proof: merkleProof, numberOfLeaves, leafIndex } = await proofDetails);
+        console.log(`Merkle proof details`)
+        console.log(`\tmerkleProof: ${merkleProof}`);
+        console.log(`\tnumberOfLeaves: ${numberOfLeaves}`);
+        console.log(`\tleafIndex: ${leafIndex}`);
+    } catch (error) {
+        console.error('RPC failed:', error);
```
Comment (95DDB) [inline app/src/app.js:93]: Perhaps we can be explicit here about the type of events/errors and events associated to success/errors ? Just for "documentation" purposes, no need to handle them explicitly here

File: app/src/register_vk.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    const session = await zkVerifySession.start()
+        .Custom(process.env.ZKV_RPC_URL)
+        .withAccount(process.env.ZKV_SEED_PHRASE);
+
+    // Send verification key to zkVerify for registration
+    const { transactionResult } = await session.registerVerificationKey()
+        .groth16()
+        .execute(vk);
+    const { statementHash } = await transactionResult;
+    console.log(`vk hash: ${statementHash}`)
```
Comment (95DDB) [inline app/src/register_vk.js:19]: Perhaps we can be explicit here about the type of events/errors and events associated to success/errors ? Just for "documentation" purposes, no need to handle them explicitly here

**Review by 95DDB — APPROVED** [review]
Great work !
Would be nice to add a Readme as well and something also in the zkVerify official docs

**Comment by 95DDB** [comment]
Might be worth to add some tests as well if possible ?

**Review by la10736 — APPROVED** [review]

**Pull request merged**